src/modules/peek/Peek.Common/Extensions/TaskExtension.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Peek.Common.Extensions
+{
+    using System;
+    using System.Threading.Tasks;
+
+    public static class TaskExtension
+    {
+        public static Task<bool> RunSafe(Func<Task> work)
```

Comment:
nit: maybe add some documentation on this RunSafe just to make it clear that it will return true if no exception and false if it caught any.